playground/engine.py

```diff
@@ -0,0 +1,818 @@
+
+        self.registered_materials = self.register_default_materials(device)  # str -> PBRMaterial
+
+    def register_available_assets(self, assets_folder):
+        available_assets = {Path(asset).stem.capitalize(): os.path.join(assets_folder, asset)
+                            for asset in os.listdir(assets_folder)
+                            if Path(asset).suffix in Primitives.SUPPORTED_MESH_EXTENSIONS}
+        # Procedural shapes are added manually
+        available_assets['Quad'] = None
+        return available_assets # i.e. {MeshName: /path/to/mesh.glb}
+
+    def register_default_materials(self, device):
+        checkboard_res = 512
+        checkboard_square = 20
+        checkboard_texture = torch.tensor([0.25, 0.25, 0.25, 1.0],
+                                          device=device, dtype=torch.float32).repeat(checkboard_res, checkboard_res, 1)
+        for i in range(checkboard_res // checkboard_square):
+            for j in range(checkboard_res // checkboard_square):
+                start_x = (2 * i + j % 2) * checkboard_square
+                end_x = min((2 * i + 1 + j % 2) * checkboard_square, checkboard_res)
+                start_y = j * checkboard_square
+                end_y = min((j + 1) * checkboard_square, checkboard_res)
+                checkboard_texture[start_y:end_y, start_x:end_x, :3] = 0.5
+        default_materials = dict(
+            solid=PBRMaterial(
+                material_id=0,
+                diffuse_map=torch.tensor([130 / 255.0, 193 / 255.0, 255 / 255.0, 1.0],
+                                         device=device, dtype=torch.float32).expand(2, 2, 4),
+                diffuse_factor=torch.ones(4, device=device, dtype=torch.float32),
+                emissive_factor=torch.zeros(3, device=device, dtype=torch.float32),
+                metallic_factor=0.0,
+                roughness_factor=0.0,
+                transmission_factor=0.0,
+                ior=1.0
+            ),
+            checkboard=PBRMaterial(
+                material_id=1,
+                diffuse_map=checkboard_texture.contiguous(),
+                diffuse_factor=torch.ones(4, device=device, dtype=torch.float32),
+                emissive_factor=torch.zeros(3, device=device, dtype=torch.float32),
+                metallic_factor=0.0,
+                roughness_factor=0.0,
+                transmission_factor=0.0,
+                ior=1.0
+            )
+        )
+        return default_materials
+
+    def set_mesh_scale_to_scene(self, mesh, transform):
+        """
+        Uses heuristics to scale the mesh so it appears nice:
+        1) mesh rescaled to unit size
+        2) if the scene is small, the mesh is rescaled to SCALE_OF_NEW_MESH_TO_SMALL_SCENE of the scene scale
+        """
+        mesh_scale = ((mesh.vertices.max(dim=0)[0] - mesh.vertices.min(dim=0)[0]).cpu()).to(transform.device)
+        transform.scale(1.0 / mesh_scale.max())
+        if self.scene_scale.max() > 5.0:    # Don't scale for large scenes
```

Comment:
sneaky logic in the primitive class is likely to confuse future users, I would prefer to avoid such heuristics inside the primitive class, and instead require to caller to chose the policy explicitly when constructing the class